cosmwasm/union-ibc/src/contract.rs

```diff
@@ -1477,6 +1480,25 @@ fn save_connection(
     Ok(())
 }
 
+fn create_channel(
+    deps: DepsMut,
+    owner: Addr,
+    channel_id: u32,
+    channel: &Channel,
+) -> Result<(), ContractError> {
+    CHANNEL_OWNER.save(deps.storage, channel_id, &owner)?;
+    CONTRACT_CHANNELS.update(deps.storage, owner, |v| -> Result<_, ContractError> {
+        Ok(match v {
+            Some(mut set) => {
+                set.insert(channel_id);
```

Comment:
what happens if the channel already exists? should check if the call to insert returns `Some`

Reply:
it's impossible as the `create` variant is only called on `init` and `try`